src/main/java/net/neoforged/neoforge/event/entity/living/ProjectileWeaponShootEvent.java

```diff
@@ -24,7 +24,7 @@ public abstract class ProjectileWeaponShootEvent extends LivingEvent {
     private final ItemStack weaponItem;
     private final ItemStack projectileItem;
 
-    public ProjectileWeaponShootEvent(LivingEntity entity, @Nullable LivingEntity target, ItemStack weaponItem, ItemStack projectileItem) {
+    ProjectileWeaponShootEvent(LivingEntity entity, @Nullable LivingEntity target, ItemStack weaponItem, ItemStack projectileItem) {
```

Comment:
I would say just make the constructor public and add the annotation `@ApiStatus.Internal`